src/VisualStudio/Core/Def/Implementation/TableDataSource/AbstractTable.cs

```diff
@@ -93,6 +94,6 @@ protected void AddTableSource(ITableDataSource source)
 
         internal ITableManager TableManager { get; }
 
-        internal abstract IReadOnlyCollection<string> Columns { get; }
+        internal abstract ImmutableArray<string> Columns { get; }
```

Comment:
⚠️ Some of the cases where we used `IReadOnlyCollection<T>` originally were intended to avoid repeat boxing of `ImmutableArray<T>`. Have you verified this does not apply to any such changes in this PR?

Reply:
yes, i have.  i've either done the boxing once.  or i've checked that hte use case (like with Table) is such that this happens rarely.  AddTableSource for example, is a one-time thing.